runtime/src/iree/base/loop_emscripten.js

```diff
@@ -0,0 +1,48 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// This is the JavaScript side of loop_emscripten.c
+//
+// References:
+//   * https://emscripten.org/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html
+//   * https://github.com/evanw/emscripten-library-generator
+//   * https://github.com/emscripten-core/emscripten/tree/main/src
+
+const LibraryLoopEmscripten = {
+  $loop_emscripten_support__postset: 'loop_emscripten_support();',
+  $loop_emscripten_support: function() {
+    class LoopEmscripten {
+      constructor() {
+        // TODO(scotttodd): store state here
+      }
```

Comment:
ES6 classes seem to work just fine, though all of the examples I looked at in Emscripten itself use the older style of declaring "classes":

```js
var MyClass = function() {
  // this is the constructor
}

MyClass.prototype.my_function = function(my_argument) { /* ... */ }
```

I actually wrote in that style at first, then VSCode gave me a "quick fix" to convert to an ES6 class :P

Reply:
hah :P given that wasm and webgpu both require contemporary browsers it's probably fine to use ES6 stuff